rust/lance-linalg/src/simd/f32.rs

```diff
@@ -249,4 +498,23 @@ mod tests {
             format!("{:?}", simd_power)
         );
     }
+
+    #[test]
+    fn test_basic_f32x16_ops() {
```

Comment:
It's probably tedious but given how technical these methods are, it might be a good idea to add tests for all the methods (e.g. add, add assign, sub, etc.)

Reply:
sure i can add some